File: CodeWars/JavaScript/MoneyMoneyMoney.js
function calculateYears(principal, interest, tax, desired) {
   let interestValue = 0;
   let contYears = 0;
    while(principal < desired){

        interestValue = principal * interest;

        principal += (interestValue.toFixed(2) - (interestValue.toFixed(2) * tax))

        contYears++;
    }

    return contYears;

}

File: CodeWars/Java/MoneyMOneyMoney.java
public class Money {
  public static int calculateYears(double principal, double interest,  double tax, double desired) {
    double moneyCurrent = principal;
    int years = 0;
    
    
    while(moneyCurrent < desired) {
      double moneyInterest = moneyCurrent * interest;
      double moneyTax = moneyInterest * tax;
      moneyCurrent += (moneyInterest - moneyTax);
      years++;
    }
    
    return years;
  }
}
